lib/rules/no-jquery-methods.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileOverview Disallow the use of specific jQuery methods.
+ * @author Marc Lynch
+ */
+'use strict';
+
+/**
+ * Determines if this expression matches the method blacklist invocation.
+ * @param {ASTNode} node The identifier node.
+ * @returns {Boolean} Returns true if the expression matches, otherwise false.
+ */
+
+const BLACKLIST = [
+  'add', 'addBack', 'after', 'ajaxComplete', 'ajaxError', 'ajaxSend', 'ajaxStart',
+  'ajaxStop', 'ajaxSuccess', 'andSelf', 'before', 'bind', 'change', 'clearQueue',
+  'clone', 'contents', 'contextmenu', 'dblclick', 'delay', 'delegate', 'dequeue', 'detach',
+  'end', 'error', 'fadeIn', 'fadeTo', 'fadeToggle', 'finish', 'focusin', 'focusout', 'hover',
+  'insertAfter', 'insertBefore', 'keydown', 'keypress', 'keyup', 'last', 'load', 'mousedown',
+  'mouseenter', 'mouseleave', 'mousemove', 'mouseout', 'mouseover', 'mouseup', 'next',
+  'nextAll', 'nextUntil', 'not', 'parentsUntil', 'prepend', 'prependTo', 'prev', 'revUntil',
+  'promise', 'queue', 'removeData', 'removeProp', 'replaceAll', 'replaceWith', 'resize', 'scroll',
+  'scrollLeft', 'select', 'serialize', 'show', 'size', 'slice', 'slideDown', 'slideToggle', 'stop',
+  'submit', 'toArray', 'toggle', 'toggleClass', 'unbind', 'undelegate', 'unload', 'unwrap', 'wrap',
+  'wrapAll', 'wrapInner'
+];
+
+function isJQueryCaller(node, name = null) {
+  let oName = node.object && node.object.callee && node.object.callee.object && node.object.callee.object.name === 'Ember';
+  let pName = node.object && node.object.callee && node.object.callee.property && node.object.callee.property.name === '$';
+  let cName = node.object && node.object.callee && node.object.callee.name === '$';
+  let lName = node.object && node.object.name === name;
+
+  return (oName) || (pName) || (cName) || (lName) ? true : false;
+}
+
+module.exports = {
+  docs: {
+    category: 'Best Practices',
+    recommended: true
+  },
+  meta: {
+    blacklist: BLACKLIST
+  },
+  create(context) {
+    let isJQuerySource = null;
+    let jQueryLocalName = null;
+
+    return {
+      ImportDeclaration(node) {
+        return isJQuerySource = node.source && node.source.value === 'jquery';
+      },
+      ImportDefaultSpecifier(node) {
+        return jQueryLocalName = isJQuerySource ? node.local.name : null;
+      },
+      VariableDeclarator(node) {
+        return jQueryLocalName = jQueryLocalName === node.init.name ? node.id.name : null;
+      },
+      MemberExpression(node) {
+        let propertyName = node.property.name;
+        let blackListName = BLACKLIST.includes(propertyName) ? propertyName : false;
+        let isThisExpression = node.object && node.object.type === 'ThisExpression';
+
+        if (!isThisExpression && blackListName && isJQueryCaller(node, jQueryLocalName)) {
+          context.report({ node: node.property, message: `The use of ${blackListName} method has been deprecated.` });
```

Comment:
It appears when a message is passed into `context.report` it is included within `meta.message` thus doing above. To confirm in the correlating test I am grabbing `rule.meta.message`.

I intentionally didn't do this because the message we are reporting is contextual and not static (blackListName is included in the message).